### PR TITLE
Extended widgetTypes access to customer user authority

### PR DESCRIPTION
**Pull request by dashevchenko**
## Pull Request description

Extended widgetTypes access to customer user authority

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [ ] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [ ] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [ ] If new yml property was added: make sure a description is added (above or near the property).

### CODE REVIEW

File: application/src/test/java/org/thingsboard/server/controller/WidgetTypeControllerTest.java [hunk 1]
```diff
@@ -190,6 +190,20 @@ public void testGetBundleWidgetTypes() throws Exception {
         Collections.sort(loadedWidgetTypes, idComparator);
 
         Assert.assertEquals(widgetTypes, loadedWidgetTypes);
+
+        loginCustomerUser();
```
Comment (smatvienko-tb) [inline application/src/test/java/org/thingsboard/server/controller/WidgetTypeControllerTest.java:194]: The test does not cover  `SYS_ADMIN` authority.

File: application/src/main/java/org/thingsboard/server/controller/WidgetTypeController.java [hunk 2]
```diff
@@ -248,7 +248,7 @@ public List<WidgetTypeDetails> getBundleWidgetTypesDetailsByBundleAlias(
 
     @ApiOperation(value = "Get all Widget types details for specified Bundle (getBundleWidgetTypes)",
             notes = "Returns an array of Widget Type Details objects that belong to specified Widget Bundle." + WIDGET_TYPE_DETAILS_DESCRIPTION + " " + SYSTEM_OR_TENANT_AUTHORITY_PARAGRAPH)
-    @PreAuthorize("hasAnyAuthority('SYS_ADMIN', 'TENANT_ADMIN')")
+    @PreAuthorize("hasAnyAuthority('SYS_ADMIN', 'TENANT_ADMIN', 'CUSTOMER_USER')")
```
Comment (smatvienko-tb) [inline application/src/main/java/org/thingsboard/server/controller/WidgetTypeController.java:251]: can't find a test case related to `/widgetTypesDetails`

**Review by smatvienko-tb — APPROVED** [review]
It looks good to me. 

Getting widget types for a customer sounds reasonable.

File: application/src/test/java/org/thingsboard/server/controller/WidgetTypeControllerTest.java [hunk 3]
```diff
@@ -190,6 +190,20 @@ public void testGetBundleWidgetTypes() throws Exception {
...
+
+        List<WidgetType> loadedWidgetTypes2 = doGetTyped("/api/widgetTypes?widgetsBundleId={widgetsBundleId}",
```
Comment (smatvienko-tb) [inline application/src/test/java/org/thingsboard/server/controller/WidgetTypeControllerTest.java:196]: loadedWidgetTypes2 -> loadedWidgetTypesCustomer

File: application/src/test/java/org/thingsboard/server/controller/WidgetTypeControllerTest.java [hunk 4]
```diff
@@ -190,6 +190,20 @@ public void testGetBundleWidgetTypes() throws Exception {
...
+                new TypeReference<>(){}, widgetsBundle.getId().getId().toString());
+        Collections.sort(loadedWidgetTypes2, idComparator);
+        Assert.assertEquals(widgetTypes, loadedWidgetTypes2);
+
+        List<WidgetTypeDetails> loadedWidgetTypes3 = doGetTyped("/api/widgetTypesDetails?widgetsBundleId={widgetsBundleId}",
```
Comment (smatvienko-tb) [inline application/src/test/java/org/thingsboard/server/controller/WidgetTypeControllerTest.java:201]: Using numbers in the name is a bad practice.
loadedWidgetTypes3 - > loadedWidgetTypeDetailsCustomer

Comment (smatvienko-tb) [reply application/src/test/java/org/thingsboard/server/controller/WidgetTypeControllerTest.java:201]: Please, add a similar test for `/api/widgetTypesDetails` above for the tenant_admin

File: application/src/main/java/org/thingsboard/server/controller/WidgetTypeController.java (hunk 2 above)
Comment (smatvienko-tb) [reply application/src/main/java/org/thingsboard/server/controller/WidgetTypeController.java:251]: Ok. I added another comment to align the test cases.

File: application/src/test/java/org/thingsboard/server/controller/WidgetTypeControllerTest.java (hunk 1 above)
Comment (smatvienko-tb) [reply application/src/test/java/org/thingsboard/server/controller/WidgetTypeControllerTest.java:194]: Did we try to test access without authority?

**Review by YevhenBondarenko — APPROVED** [review]

**Pull request merged**